### PR TITLE
[Router] Fix namespace sync

### PR DESCRIPTION
**Pull request by marun**
As per the referenced bz, a router filtering namespaces by label is not exposing routes from a newly-labeled namespace after the resync interval.

Reference: https://bugzilla.redhat.com/show_bug.cgi?id=1415112

cc: @openshift/networking

### CODE REVIEW

**Comment by marun** [comment]
So far I have a test that reproduces the problem. Still working on the fix.

**Comment by marun** [comment]
While investigating this, I noticed that there is a potential for timer skew that could delay route admission: #13436 

I don't think that is the problem here, though.

**Comment by marun** [comment]
The problem is the empty ``EventType``.  A namespace sync adds the namespace just fine, but then the events from the subsequent route sync are ignored because they have an empty event.

File: pkg/client/cache/eventqueue.go [hunk 1]
```diff
@@ -290,8 +290,16 @@ func (eq *EventQueue) Pop() (watch.EventType, interface{}, error) {
 		key := eq.queue[0]
 		eq.queue = eq.queue[1:]
 
-		eventType := eq.events[key]
-		delete(eq.events, key)
+		eventType, ok := eq.events[key]
+		if ok {
+			delete(eq.events, key)
+		} else {
+			// The event type has been removed by a previous call to
```
Comment (marun) [inline pkg/client/cache/eventqueue.go]: @smarterclayton This looks like a long-standing issue, not sure why it hasn't been noticed before.

**Comment by knobunc** [comment]
Thanks Maru!

**Comment by knobunc** [comment]
[test]

File: pkg/client/cache/eventqueue.go (hunk 1 above)
Comment (smarterclayton) [inline pkg/client/cache/eventqueue.go]: Yes, it's possible that this is the only use of event queue in a meaningful form left in the code.

**Comment by smarterclayton** [comment]
Lgtm as well

**Comment by knobunc** [comment]
[merge]

**Pull request merged**